### PR TITLE
Fix calico RKE2 Calico release notes

### PR DESCRIPTION
**Pull request by johnatasr**
#### Proposed Changes ####

- Created a function to treat Calico's url in conformity with Tigera.io lastest changes for release notes
- RKE2 template changes

#### Types of Changes ####

- backend code 

#### Verification ####

1 - Compile this ecm-distro-tools branch
2 - Execute `./cmd/gen_release_notes/bin/gen_release_notes -r rke2 -m v1.27.4+rke2r1 -p v1.27.3+rke2r1`
3 - Must show like this:

```
## Packaged Component Versions
| Component | Version |
| --- | --- |
| Kubernetes | [v1.27.4](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#v1274) |
| Etcd | [v3.5.7-k3s1](https://github.com/k3s-io/etcd/releases/tag/v3.5.7-k3s1) |
| Containerd | [v1.7.1-k3s1](https://github.com/k3s-io/containerd/releases/tag/v1.7.1-k3s1) |
| Runc | [v1.1.7](https://github.com/opencontainers/runc/releases/tag/v1.1.7) |
| Metrics-server | [v0.6.3](https://github.com/kubernetes-sigs/metrics-server/releases/tag/v0.6.3) |
| CoreDNS | [v1.10.1](https://github.com/coredns/coredns/releases/tag/v1.10.1) |
| Ingress-Nginx | [4.6.1](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.6.1) |
| Helm-controller | [v0.15.2](https://github.com/k3s-io/helm-controller/releases/tag/v0.15.2) |

### Available CNIs
| Component | Version | FIPS Compliant |
| --- | --- | --- |
| Canal (Default) | [Flannel v0.22.0](https://github.com/k3s-io/flannel/releases/tag/v0.22.0)<br/>[Calico v3.25.1](https://projectcalico.docs.tigera.io/archive/v3.25/release-notes/#v3.25.1) | Yes |
| Calico | [v3.26.1](https://docs.tigera.io/calico/latest/release-notes/#v3.26) | No |
| Cilium | [v1.13.2](https://github.com/cilium/cilium/releases/tag/v1.13.2) | No |
| Multus | [v4.0.2](https://github.com/k8snetworkplumbingwg/multus-cni/releases/tag/v4.0.2) | No |

```

#### Testing ####

Local test
ecmcalico

#### Linked Issues ####

https://github.com/rancher/ecm-distro-tools/issues/231

### CODE REVIEW

File: release/release.go [hunk 1]
```diff
@@ -45,7 +45,9 @@ type rke2ReleaseNoteData struct {
 	HelmControllerVersion string
 	FlannelVersion        string
 	CanalCalicoVersion    string
+	CanalCalicoUrl        string
```
Comment (briandowns) [inline release/release.go:48]:
```suggestion
	CanalCalicoURL        string
```

File: release/release.go [hunk 2]
```diff
@@ -45,7 +45,9 @@ type rke2ReleaseNoteData struct {
...
 	CalicoVersion         string
+	CalicoUrl             string
```
Comment (briandowns) [inline release/release.go:50]:
```suggestion
	CalicoURL             string
```

File: release/release.go [hunk 3]
```diff
@@ -536,25 +540,56 @@ func sqliteVersionBinding(sqliteVersion string) string {
...
+		return "https://docs.tigera.io/calico/latest/release-notes/#" + formattedVersion
+	}
+
+	return calicoArchiveURL
+}
+
 // findInURL will get and scan a url to find a slice submatch for all the words that matches a regex
 // if the regex is empty then it will return the lines in a file that matches the str
-func findInURL(url, regex, str string) []string {
+func findInURL(url, regex, str string, checkStatusCode bool) []string {
```
Comment (briandowns) [inline release/release.go:584]: It doesn't appear as though the bool is ever not true so I don't think it's necessary to have as an argument.

**Review by briandowns — APPROVED** [review]

**Review by Tashima42 — APPROVED** [review]
works

**Pull request merged**